src/lib/sql/sql_translator.cpp

```diff
@@ -1834,15 +1835,25 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_import(const hsql::Im
   // so on. Anyway, we would need to decouple data loading and storing the table and have to load metadata or even the
   // whole table when translating the SQL statement.
   AssertInput(!import_statement.whereClause, "Predicates on imported files are not supported.");
+
+  auto encoding = std::optional<EncodingType>{};
+  if (import_statement.encoding) {
+    encoding = magic_enum::enum_cast<EncodingType>(import_statement.encoding);
+    AssertInput(encoding, "Unknown encoding type '" + std::string{import_statement.encoding} + "'.");
+  }
+
   return ImportNode::make(import_statement.tableName, import_statement.filePath,
-                          import_type_to_file_type(import_statement.type));
+                          import_type_to_file_type(import_statement.type), encoding);
 }
 
 // NOLINTNEXTLINE: while this particular method could be made static, others cannot.
 std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_export(const hsql::ExportStatement& export_statement) {
   auto sql_identifier_resolver = std::make_shared<SQLIdentifierResolver>();
   auto lqp = std::shared_ptr<AbstractLQPNode>{};
 
+  AssertInput(!export_statement.encoding,
```

Comment:
I don't understand how the second sentence is relevant here?

Reply:
I mean, why should you want to export a table with a specific encoding? Only for the case that this encoding is applied when you import table again somewhere. And for this, you can specify the encoding directly during the import.